apps/console/src/features/users/components/user-profile.tsx

```diff
@@ -229,63 +234,128 @@ export const UserProfile: FunctionComponent<UserProfilePropsInterface> = (
         if (!isEmpty(profileSchema) && !isEmpty(userInfo)) {
             const tempProfileInfo: Map<string, string> = new Map<string, string>();
 
-            proSchema.forEach((schema: ProfileSchemaInterface) => {
-                const schemaNames = schema.name.split(".");
-
-                if (schemaNames.length === 1) {
-                    if (schemaNames[0] === "emails") {
-                        if(ProfileUtils.isStringArray(userInfo[schemaNames[0]])) {
-                            const emails: any[] = userInfo[schemaNames[0]];
-                            const primaryEmail = emails.find((subAttribute) => typeof subAttribute === "string");
-
-                            // Set the primary email value.
-                            tempProfileInfo.set(schema.name, primaryEmail);
+            if (adminUserType === "internal") {
+                proSchema.forEach((schema: ProfileSchemaInterface) => {
+                    const schemaNames = schema.name.split(".");
+    
+                    if (schemaNames.length === 1) {
+                        if (schemaNames[0] === "emails") {
```

Comment:
```suggestion
                        if (schemaNames[0] === "emails") {
```

What is `0` here? What are we accessing? Can we create a const and allocate the index. So we can avoid magic indexes. Check other changes with indexes as well..

`const SOMETHING = 0;` then `schemaNames[SOMETHING] === "emails"`